public/app/plugins/datasource/cloudwatch/components/LogGroupSelector.tsx

```diff
@@ -0,0 +1,163 @@
+import { debounce, intersection, unionBy } from 'lodash';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { SelectableValue, toOption } from '@grafana/data';
+import { MultiSelect } from '@grafana/ui';
+import { InputActionMeta } from '@grafana/ui/src/components/Select/types';
+import { notifyApp } from 'app/core/actions';
+import { createErrorNotification } from 'app/core/copy/appNotification';
+import { dispatch } from 'app/store/store';
+
+import { CloudWatchDatasource } from '../datasource';
+import { appendTemplateVariables } from '../utils/utils';
+
+const MAX_LOG_GROUPS = 20;
+const MAX_VISIBLE_LOG_GROUPS = 4;
+const DEBOUNCE_TIMER = 300;
+
+export interface LogGroupSelectorProps {
+  region: string;
+  selectedLogGroups: string[];
+  onChange: (logGroups: string[]) => void;
+
+  datasource?: CloudWatchDatasource;
+  allowCustomValue?: boolean;
+  onRunQuery?: () => void;
+  onOpenMenu?: () => Promise<void>;
+  refId?: string;
+  width?: number | 'auto';
+  saved?: boolean;
+}
+
+export const LogGroupSelector: React.FC<LogGroupSelectorProps> = ({
+  region,
+  selectedLogGroups,
+  onChange,
+  datasource,
+  allowCustomValue,
+  onRunQuery,
+  onOpenMenu,
+  refId,
+  width,
+  saved = true,
+}) => {
+  const [loadingLogGroups, setLoadingLogGroups] = useState(false);
+  const [availableLogGroups, setAvailableLogGroups] = useState<Array<SelectableValue<string>>>([]);
+  const logGroupOptions = useMemo(
+    () => unionBy(availableLogGroups, selectedLogGroups?.map(toOption), 'value'),
+    [availableLogGroups, selectedLogGroups]
+  );
+
+  const fetchLogGroupOptions = useCallback(
+    async (region: string, logGroupNamePrefix?: string) => {
+      if (!datasource) {
+        return [];
+      }
+      try {
+        const logGroups: string[] = await datasource.describeLogGroups({
+          refId,
+          region,
+          logGroupNamePrefix,
+        });
+        return logGroups.map(toOption);
+      } catch (err) {
+        let errMessage = 'unknown error';
+        if (typeof err !== 'string') {
+          try {
+            errMessage = JSON.stringify(err);
+          } catch (e) {}
+        } else {
+          errMessage = err;
```

Comment:
this error handling seems a bit weird, can you comment on it? Also, it seems that it will never be 'unknown error' based in the logic so you can just do `let errMessage = err` at the beginning.

Reply:
Totally agree! However, I know that this logic was not introduced by you Isabella so might be hard to comments on this. If you want to, you can create another issue for "revamping error handling for meta data queries" in the cloudwatch plugin and then we do a full rewrite of this another time.

Reply:
Andrés' comment about `let errMessage = err` would be nice to apply now though!

Reply:
changed to to `let errMessage = err`, I'm still not sure why the error handling is like this, but I filed a follow-up issue (https://github.com/grafana/grafana/issues/50893) I'll add a link to the line once this goes in.

Reply:
I had to change this back in order for it to lint correctly